docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,207 @@ Example for the `__time` dimension:
+```json
+"parseSpec": {
+  "type": "customJson",
+  "keyFieldName": "key",
+  "valueFieldName": "value"
+}
+```
+
+
+##### simpleJson namespaceParseSpec
+The `simpleJson` namespaceParseSpec does not take any parameters. It is simply a line delimited json file where the field is the key, and the field's value is the value.
+
+*example input*
+ 
+```json
+{"foo": "bar"}
+{"baz": "bat"}
+{"buck": "truck"}
+```
+
+*example namespaceParseSpec*
+```json
+"namespaceParseSpec":{
+  "type": "simpleJson"
+}
+```
+
+
+#### JDBC namespace update
+The JDBC namespaces will pull from a database cache. If the `tsColumn` is set it must be able to accept comparisons in the format `'2015-01-01 00:00:00'`. For example, the following must be valid sql for the table `SELECT * FROM some_namespace_table WHERE timestamp_column >  '2015-01-01 00:00:00'`. If `tsColumn` is set, the caching service will attempt to only pull values that were written *after* the last sync. If `tsColumn` is not set, the entire table is pulled every time.
+
+```json
+{
+  "namespace":"some_namespace",
+  "connectorConfig":{
+    "createTables":true,
+    "connectURI":"jdbc:mysql://localhost:3306/druid",
+    "user":"druid",
+    "password":"diurd"
+  },
+  "table":"some_namespace_table",
+  "keyColumn":"the_old_dim_value",
+  "valueColumn":"the_new_dim_value",
+  "tsColumn":"timestamp_column",
+  "updateMs":600000
+}
+```
+
+### Kafka namespace update (Experimental)
+It is possible to plug into a kafka topic whose key is the old value and message is the desired new value (both in UTF-8). This requires the following extension: "io.druid.extensions:druid-dim-rename-kafka8"
+```json
+{
+  "type":"kafka",
+  "namespace":"testTopic",
+  "kafkaTopic":"testTopic"
+}
+```
+#### Kafka renames
+The extension `druid-dim-rename-kafka8` enables reading from a kafka feed which has name/key pairs to allow renaming of dimension values. An example use case would be to rename an ID to a human readable format.
+
+Currently the historical node caches the key/value pairs from the kafka feed in an ephemeral memory mapped DB via MapDB.
+
+#### Configuration
+The following options are used to define the behavior:
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.query.rename.kafka.properties`|A json map of kafka consumer properties. See below for special properties.|See below|
+
+The following are the handling for kafka consumer properties in `druid.query.rename.kafka.properties`
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`zookeeper.connect`|Zookeeper connection string|`localhost:2181/kafka`|
+|`group.id`|Group ID, auto-assigned for publish-subscribe model and cannot be overridden|`UUID.randomUUID().toString()`|
+|`auto.offset.reset`|Setting to get the entire kafka rename stream. Cannot be overridden|`smallest`|
+
+#### Hooking up namespaces
+To hook up a kafka topic to a namespace, a `KafkaExtractionNamespace` used in the namespace update posted to zookeeper.
```

Comment:
Have you really been far even as decided to use even go want to do look more like?

will fix